download/config.go

```diff
@@ -45,22 +45,33 @@ func NewConfig(ctx context.Context, mongoVersionStr string) (*Config, error) {
 	if versionErr != nil {
 		return nil, versionErr
 	}
-
 	downloadUrl, err := getDownloadUrl(*version)
 	if err != nil {
 		return nil, err
 	}
+	if downloadUrl == DoNotDownload {
+		basePath, err := defaultBaseCachePath()
+		if err != nil {
+			log.Error(ctx, "cache directory not found", err)
+			return nil, err
+		}
+		return &Config{
+			mongoVersion: *version,
+			cachePath:    basePath + "/mongod",
+		}, nil
+	} else {
+		cachePath, err := buildBinCachePath(ctx, downloadUrl)
+		if err != nil {
+			return nil, err
+		}
 
-	cachePath, err := buildBinCachePath(ctx, downloadUrl)
-	if err != nil {
-		return nil, err
+		return &Config{
+			mongoVersion: *version,
+			mongoUrl:     downloadUrl,
+			cachePath:    cachePath,
+		}, nil
```

Comment:
@rafahop if this is still not acceptable from you POV, then I will not merge this PR and will keep it rebased with `main`
Thanks for all the patience on this :)

Reply:
Yes, I think keeping this in a separate branch for specific use would be best. It does not really fit the purpose of the library and requires some manual steps for Manjaro so yes, I wouldn't merge it into the main branch.